### PR TITLE
periodically clean snapshot

### PR DESCRIPTION
**Pull request by doodles526**
Use etcd's fileutil package to periodically cleanup snapshots

### CODE REVIEW

File: raft.go
Comment (gyuho) [inline raft.go]: @doodles526 

Q. then shouldn't `MaxRetainedSnapshots` be specified as 3 in `DefaultSnapshotConfig`?
I see that field is empty, so 0? etcd `DefaultMaxSnapshots` is 5.

**Review by gyuho — APPROVED** [review]
lgtm.

File: raft.go
Comment (doodles526) [reply raft.go]: Good catch. Canoe's `DefaultSnapshotConfig` defaults to disabling snapshotting altogether. In general I've tried to have Canoe default to disabling features and retaining data when possible. Thus the ideal for the default would be to disable snapshotting. And when `SnapshotConfig` is specified in the config and `MaxRetainedSnapshots` is 0, then just never purge Snapshot files. At least in my opinion I would prefer a library that preserves data unless I explicitly tell it to destroy it. I've amended the commit to reflect that. So if you have an lgtm after that I'll merge in

**Comment by gyuho** [comment]
@doodles526  lgtm.

That makes sense to me. Thanks!

**Pull request merged**
